File: core/src/main/java/com/nfthub/core/repository/TagRepository.java
package com.nfthub.core.repository;

import com.nfthub.core.entity.Tag;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;

@Repository
public interface TagRepository extends JpaRepository<Tag, Long> {
    Optional<Tag> findByName(String name);

    List<Tag> findAllByNameLike(String name);


}

File: src/main/kotlin/com/nfthub/api/repository/TagRepository.kt
package com.nfthub.api.repository

import com.nfthub.api.entity.Tag
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository


@Repository
interface TagRepository : JpaRepository<Tag, Long> {
    fun findByName(name: String): Tag?
}